test/Microsoft.AspNetCore.Http.Abstractions.Tests/HostStringTest.cs

```diff
@@ -92,5 +92,37 @@ public void Ctor_CreatesFromHostAndPort(string sourceHost, int sourcePort, strin
             Assert.Equal(expectedHost, host);
             Assert.Equal(expectedPort, port);
         }
+
+        [Fact]
+        public void Equals_EmptyHostStringAndDefaultHostString()
+        {
+            // Act and Assert
+            Assert.Equal(new HostString(string.Empty), default(HostString));
+            Assert.Equal(default(HostString), new HostString(string.Empty));
+            Assert.True(new HostString(string.Empty) == default(HostString));
```

Comment:
Explicitly checking == operator. Think the last pair is redundant because assert.equal calls equals.